.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,195 @@
+        run: |
+          git log -n 5 --pretty=format:"%H" >> tag_commits.txt
+          asv run -v HASHFILE:tag_commits.txt | tee asv-output.log  
+          if grep -q failed asv-output.log; then 
+            echo "Some benchmarks have failed!"
+            exit 1
+          fi
+
+      - name: Generate Graphs and HTML
+        if: github.event_name != 'pull_request'
+        run: asv publish
+
+      - name: Delete env files
+        run: rm -r .asv/env
+
+      - name: Create results html on dest repo
+        if: github.event_name != 'pull_request'
+        continue-on-error: true
+        uses: cpina/github-action-push-to-another-repository@main
+        env:
+          API_TOKEN_GITHUB: ${{ secrets.BOT_TOKEN }}
+        with:
+          source-directory: ".asv/html"
+          destination-github-username: "tardis-sn"
+          destination-repository-name: "stardis-benchmarks"
+          user.email: tardis.sn.bot@gmail.com
+          target-branch: main
+
+      - name: Push misc data to dest repo
+        if: github.event_name != 'pull_request'
+        continue-on-error: true
+        uses: cpina/github-action-push-to-another-repository@main
+        env:
+          API_TOKEN_GITHUB: ${{ secrets.BOT_TOKEN }}
+        with:
+          source-directory: ".asv/results"
+          destination-github-username: "tardis-sn"
+          destination-repository-name: "stardis-benchmarks"
+          user.email: tardis.sn.bot@gmail.com
+          target-branch: main
+          target-directory: "misc"
+
+      - name: Compare HEAD with main if PR
+        if: github.event_name == 'pull_request'
+        continue-on-error: true # TODO: step failed sporadically while testing
+        run: |
+          asv continuous ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }}  | tee asv-cont-output.log  
+          if grep -q failed asv-cont-output.log; then 
+            echo "Some benchmarks have errors!"
+            exit 1
+          fi
+
+      - run: asv compare ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} | tee asv-compare-output.log
+      - run: asv compare ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} --only-changed | tee asv-compare-changed-output.log
+
+      - name: asv output 1
+        id: asv_pr_vs_main
+        uses: juliangruber/read-file-action@v1.0.0
+        with:
+          path: asv-compare-output.log
+
+      - name: asv output 2
+        id: asv_pr_vs_main_changed
+        uses: juliangruber/read-file-action@v1.0.0
+        with:
+          path: asv-compare-changed-output.log
+
+      - name: Find Comment
+        if: github.event_name == 'pull_request'
+        uses: peter-evans/find-comment@v1
+        id: fc
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          comment-author: "tardis-bot"
+          body-includes: I ran benchmarks as you asked
+
```

Comment:
I have made some changes to this in https://github.com/tardis-sn/tardis/pull/2330
You might need to have a look at it